examples/openshift-lightspeed.yaml

```diff
@@ -0,0 +1,161 @@
+        imagePullPolicy: Always
+        name: lightspeed-w-rag
+        env:
+          - name: OLS_CONFIG_FILE
+            value: /app-root/config/olsconfig.yaml
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+          - name: olsconfig
+            mountPath: /app-root/config
+          - name: openai
+            mountPath: /app-root/config/openai
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext: {}
+      terminationGracePeriodSeconds: 30
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: openshift-lightspeed
+    app.kubernetes.io/component: lightspeed-w-rag
+    app.kubernetes.io/instance: lightspeed-w-rag
+    app.kubernetes.io/name: lightspeed-w-rag
+    app.kubernetes.io/part-of: lightspeed-w-rag-app
+  name: lightspeed-w-rag
+spec:
+  internalTrafficPolicy: Cluster
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - name: 8080-tcp
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: openshift-lightspeed
+    deployment: lightspeed-w-rag
+  sessionAffinity: None
+  type: ClusterIP
+
+#---
+#apiVersion: route.openshift.io/v1
+#kind: Route
+#metadata:
+#  labels:
+#    app: openshift-lightspeed
+#    app.kubernetes.io/component: lightspeed-w-rag
+#    app.kubernetes.io/instance: lightspeed-w-rag
+#    app.kubernetes.io/name: lightspeed-w-rag
+#    app.kubernetes.io/part-of: lightspeed-w-rag-app
+#  name: lightspeed-w-rag
+#spec:
+#  port:
+#    targetPort: 8080-tcp
+#  tls:
+#    insecureEdgeTerminationPolicy: Redirect
+#    termination: edge
+#  to:
+#    kind: Service
+#    name: lightspeed-w-rag
+#    weight: 100
+#  wildcardPolicy: None
+
+---
+apiVersion: v1
+data:
+  olsconfig.yaml: |
+    llm_providers:
+      - name: openai
+        url: "https://api.openai.com/v1"
+        credentials_path: config/openai/openai_api_key.txt
+        models:
+          - name: gpt-4-1106-preview
+          - name: gpt-3.5-turbo
+    ols_config:
+      conversation_cache:
+        type: memory
+        memory:
+          max_entries: 1000
+      logging_config:
+        app_log_level: info
+        lib_log_level: warning
+      default_provider: openai
+      default_model: gpt-4-1106-preview
+    dev_config:
+      enable_dev_ui: true
+      # llm_temperature_override: 0
+      # disable_question_validation: false
+      # disable_auth: false
+immutable: false
+kind: ConfigMap
+metadata:
+  labels:
+    app: openshift-lightspeed
+    app.kubernetes.io/component: lightspeed-w-rag
+    app.kubernetes.io/instance: lightspeed-w-rag
+    app.kubernetes.io/name: lightspeed-w-rag
+    app.kubernetes.io/part-of: lightspeed-w-rag-app
+  name: olsconfig
+
+---
+apiVersion: v1
+data:
+  openai_api_key.txt: <BASE64_ENCODED_OPENAI_API_KEY>
```

Comment:
using the stringData field would make this easier for people so they don't have to base64 encode.

![image](https://github.com/openshift/lightspeed-service/assets/4974046/a1be8d2b-e5c9-4844-8e80-9fa02861272b)